src/main/java/nextstep/member/domain/GithubClient.java

```diff
@@ -0,0 +1,76 @@
+package nextstep.member.domain;
+
+
+import nextstep.member.domain.dto.GithubAccessTokenRequest;
+import nextstep.member.domain.dto.GithubAccessTokenResponse;
+import nextstep.member.domain.dto.GithubProfileResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class GithubClient {
+
+    private String clientId;
+    private String clientSecret;
+    private String tokenRequestUri;
+    private String profileUri;
+
+    public GithubClient(
+            @Value("${security.github.token.client-id}")
+            String clientId,
+            @Value("${security.github.token.client-secret}")
+            String clientSecret,
+            @Value("${security.github.token.request-uri}")
+            String tokenRequestUri,
+            @Value("${security.github.profile-uri}")
+            String profileUri
+    ) {
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+        this.tokenRequestUri = tokenRequestUri;
+        this.profileUri = profileUri;
+    }
+
+    public String getAccessToken(String code) {
+        return requestToken(code);
+    }
+
+    private String requestToken(String code) {
+        GithubAccessTokenRequest githubAccessTokenRequest = new GithubAccessTokenRequest(
+                code,
+                clientId,
+                clientSecret
+        );
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.add("Accept", MediaType.APPLICATION_JSON_VALUE);
```

Comment:
이미 정의된 상수를 사용해볼 수도 있을 것 같습니다.
```suggestion
        headers.add(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE);
```

Reply:
[94c0ab9](https://github.com/next-step/atdd-subway-favorite/pull/696/commits/94c0ab9145cfd4ff20eaa185e26c3bc017ea7143) 에 적용했습니다~

이런 정보를 알려주셔서 너무 감사합니다 🥹